rotkehlchen/start.py

```diff
@@ -0,0 +1,96 @@
+    process: The process Popen instance that is started by the orchestrator."""
+    entrypoint: Callable
+    process: Popen | None
+
+    def __init__(self, entrypoint: Callable, process: Popen | None) -> None:
+        self.entrypoint = entrypoint
+        self.process = process
+
+
+class ProcessesManager:
+    """Manages the processes that are started by the orchestrator."""
+    def __init__(self) -> None:
+        self.processes: dict[ProcessType, ProcessData] = {
+            ProcessType.API_SERVER: ProcessData(
+                entrypoint=self.start_api_server,
+                process=None,
+            ),
+            ProcessType.DB_WRITER: ProcessData(
+                entrypoint=self.start_db_writer,
+                process=None,
+            ),
+        }
+
+    def start_api_server(self) -> None:
+        from rotkehlchen. __main__ import main as backend  # isort:skip  # pylint: disable=import-outside-toplevel  # to make the orchestrator lighter
+        backend()
+
+    def start_db_writer(self) -> None:
+        from rotkehlchen.db.drivers.server import main as db_writer   # isort:skip  # pylint: disable=import-outside-toplevel  # to make the orchestrator lighter
+        db_writer()
+
+    def shutdown(self, signum: int, frame: FrameType | None = None) -> None:  # pylint: disable=unused-argument
+        for process_data in self.processes.values():
+            if process_data.process is not None:
+                process_data.process.send_signal(signum)
+                process_data.process.terminate()
+
+        sys.exit(0)
+
+
+def main() -> None:
+    manager = ProcessesManager()
+
+    signal.signal(signal.SIGINT, manager.shutdown)
+    if os.name != 'nt':
+        signal.signal(signal.SIGTERM, manager.shutdown)
+        signal.signal(signal.SIGQUIT, manager.shutdown)
+    else:
+        # Handle the windows control signal as stated here: https://pyinstaller.org/en/stable/feature-notes.html#signal-handling-in-console-windows-applications-and-onefile-application-cleanup  # noqa: E501
+        # This logic handles the signal sent from the bootloader equivalent to sigterm in
+        # addition to the signals sent by windows's taskkill.
+        # Research documented in https://github.com/yabirgb/rotki-python-research
+        import win32api  # pylint: disable=import-outside-toplevel  # isort:skip
+        win32api.SetConsoleCtrlHandler(manager.shutdown, True)
+
+    args = app_args(
+        prog='rotki orchestrator',
+        description='The rotki orchestrator to run other rotki processes.',
+    ).parse_args()
+
+    if args.process == 'all':
+        executable = [sys.executable] + sys.orig_argv[1:]
+        for process_type, process_data in manager.processes.items():
+            process_data.process = Popen(executable + ['--process', process_type.value])  # noqa: S603
+
+        if (backend_process := manager.processes[ProcessType.API_SERVER].process) is not None:
+            backend_process.wait()
```

Comment:
One thing that I noticed is that you don't check that all the processes are running right? It can happen that the db writter fails to start and you don't check it